k8s/monocle-deployment.yml

```diff
@@ -0,0 +1,287 @@
+            - env:
+                - name: REACT_APP_API_URL
+                  value: ${MONOCLE_API_URL:-http://localhost:9876}
+                - name: REACT_APP_TITLE
+                  value: ${MONOCLE_TITLE:-}
+              image: changemetrics/monocle_web:latest
+              name: web
+              ports:
+                - containerPort: 3000
+              resources: { }
+          restartPolicy: Always
+    status: { }
+  - apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      labels:
+        io.kompose.service: api
+      name: api
+    spec:
+      replicas: 1
+      selector:
+        matchLabels:
+          io.kompose.service: api
+      template:
+        metadata:
+          creationTimestamp: null
+          labels:
+            io.kompose.service: api
+        spec:
+          containers:
+            - args:
+                - uwsgi
+                - --uid
+                - guest
+                - --gid
+                - nogroup
+                - --http
+                - :9876
+                - --socket
+                - :9877
+                - --manage-script-name
+                - --mount
+                - /app=monocle.webapp:app
+              image: changemetrics/monocle_backend:latest
+              livenessProbe:
+                exec:
+                  command:
+                    - python -c "import requests,sys;
+                      r=requests.get('http://localhost:9876/api/0/health');
+                      print(r.text);
+                      sys.exit(1) if r.status_code!=200 else sys.exit(0)"
+              name: api
+              ports:
+                - containerPort: 9876
+                - containerPort: 9877
+              resources: { }
+              volumeMounts:
+                - mountPath: /etc/monocle
+                  name: config-data
+          restartPolicy: Always
+          volumes:
+            - name: config-data
+              persistentVolumeClaim:
+                claimName: config-data
+    status: { }
+  - apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      annotations:
+        kompose.cmd: kompose convert
+        kompose.version: 1.17.0 (HEAD)
+      creationTimestamp: null
+      labels:
+        io.kompose.service: crawler
+      name: crawler
+    spec:
+      replicas: 1
+      selector:
+        matchLabels:
+          io.kompose.service: crawler
+      template:
+        metadata:
+          creationTimestamp: null
+          labels:
+            io.kompose.service: crawler
+        spec:
+          containers:
+            - args:
+                - monocle
+                - --elastic-conn
+                - elastic:9200
+                - crawler
+                - --config
+                - /etc/monocle/config.yaml
+              env:
+                - name: APP_ID
+                  value: ${GITHUB_APP_ID:-}
+                - name: APP_KEY_PATH
+                  value: ${GITHUB_APP_KEY_PATH:-/etc/monocle/app_key.rsa}
```

Comment:
These vars have to be kept as secrets, as k8s cannot parse them.
`Traceback (most recent call last):
  File "/usr/local/bin/monocle", line 33, in <module>
    sys.exit(load_entry_point('monocle==0.8.1', 'console_scripts', 'monocle')())
  File "/usr/local/lib/python3.9/site-packages/monocle-0.8.1-py3.9.egg/monocle/main.py", line 174, in main
  File "/usr/local/lib/python3.9/site-packages/monocle-0.8.1-py3.9.egg/monocle/github/application.py", line 201, in get_app
FileNotFoundError: [Errno 2] No such file or directory: '${GITHUB_APP_KEY_PATH:-/etc/monocle/app_key.rsa}'
`

Reply:
the meaning of the comment is that when migrating from docker-compose to k8s, this env value was kept as is

we assume there was Argument substitution and thus makes this issue not happen on docker-compose

the proposed solution is moving the env var to a secret resource and linking it into the contrainer
WDYT?

Reply:
Yes moving the env into a secret resources or config map seems to be the solution as far as I understand k8s.